hw1_collections/hw_collections.py

```diff
@@ -0,0 +1,89 @@
+# List of dictionaries for translating sequences #
+
+complement_dna = {'A':'T', 'T':'A', 'G':'C', 'C':'G',
+                  'a':'t', 't':'a', 'g':'c', 'c':'g'}
+
+complement_rna = {'A':'U', 'U':'A', 'G':'C', 'C':'G',
+                  'a':'u', 'u':'a', 'g':'c', 'c':'g'}
+
+transcribe_dna_to_rna = {'A':'A', 'T':'U', 'G':'G', 'C':'C',
+                         'a':'a', 't':'u', 'g':'g', 'c':'c'}
+
+transcribe_rna_to_dna = {'A':'T', 'U':'A', 'G':'C', 'C':'G',
+                         'a':'t', 'u':'a', 'g':'c', 'c':'g'}
+
+allowed_symbols = ['A', 'T', 'G', 'C', 'U',
+                   'a', 't', 'g', 'c', 'u']
+
+# List of commands #
+
+def complement(sequence):
+    if any(U in sequence for U in ['U', 'u']):
+        for nucleotide in sequence:
+            decoded.append(complement_rna[nucleotide])
+    else:
+        for nucleotide in sequence:
+            decoded.append(complement_dna[nucleotide])
+    return(''.join(decoded))
```

Comment:
В питоне не принято ставить скобочки при использовании `return`
```suggestion
    return ''.join(decoded)
```

Reply:
А за реализацию комплементации через списки и джоин лайк 👍

Reply:
А почему это хорошо? Я просто сделала как было удобнее и не слишком задумывалась, но теперь мне интересно почему так лучше делать...

Reply:
@Sarsaparella, альтернативная реализация возможна через "наращивание" пустой строки через прибавление к ней комплементированных нуклеотидов. Она не очень эффективна, так как строки неизменяемые и с каждым нуклеотидом питону приходится создавать всё более большие строки, что довольно медленно.

Reply:
Списки неизменяемы, поэтому с ними таких проблем нет